tests/cli/test_utils.py

```diff
@@ -101,3 +113,171 @@ def test_validate_with_invalid_directory_if_default_is_valid(tmp_path: pathlib.P
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_CORE,
+            "error": False,
+        },
+        {
+            "config_data": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": None,
+            },
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_CORE,
+            "error": False,
+        },
+        {
+            "config_data": {},
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS,
+            "error": True,
+        },
+        {
+            "config_data": {
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+                "imports": "other-folder",
+            },
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_NLU,
+            "error": True,
+        },
+        {
+            "config_data": None,
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_NLU,
+            "error": True,
+        },
+        {
+            "config_data": None,
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS,
+            "error": False,
+        },
+        {
+            "config_data": None,
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_CORE,
+            "error": False,
+        },
+        {
+            "config_data": None,
+            "default_config": None,
+            "mandatory_keys": CONFIG_MANDATORY_KEYS,
+            "error": True,
+        },
+    ],
+)
+def test_get_valid_config(parameters):
```

Comment:
It seems like it would be easier to read the test if it was split between three tests:
* test valid config
* test with default config
* test invalid config

That way the name of the test suggests what should happen and it puts less cognitive load on the examiner/reader of the test.

Reply:
Addressed: https://github.com/RasaHQ/rasa/pull/11978/commits/1f874275f22df4119245d1d1326de818bf041adb